lopper/assists/subsystem.py

```diff
@@ -675,15 +675,89 @@ def chosen_expand( tree, chosen_node ):
     tree.add( chosen_node_copy, merge=True )
 
 def reserved_memory_expand( tree, reserved_memory_node ):
-    # there isn't any specific processing required for reserved memory
-    # at the moment. We just copy it to the main tree as-is. If a
-    # reserved-memory node exists, they should be merged.
+    # for each domain calling into this:
+    # 1. look up to see if reserved memory already exists
+    # 2. store old nodes
+    # 3. transform domain's reserved memory list of strings to phandles
+    #     where each string matches a rserved memory entry name
+    # 4. If a reserved memory node has a start/size tuple then transform
+    #    this to a reg property
+    try:
+        res_mem_node = tree["/reserved-memory"]
+        # get reg vals from pre-existing reserved memory
+        pre_existing_res_mem_nodes = [ n for n in res_mem_node.subnodes(children_only=True) ]
+    except:
+        pre_existing_res_mem_nodes = []
+        res_mem_node = LopperNode(-1, "/reserved-memory")
+        tree.add(res_mem_node)
```

Comment:
What about the simpler cases ?  We shouldn't insist on the (slightly) more complexity of needing to define nodes for reserved memory. There are cases where we just want a simple start and size in the yaml and have that become the reserved memory node.

Reply:
@zeddii  but if a user defines start/size then why not also just add to this routine the start/size chunk handling?

this routine only gets if a user wants to have reserved-memory anyway right?

Reply:
yes. once we figure out what to do with them, we are ok. There may be existing anchor usage in yaml files, so we can't just remove the support without a warning or migration plan.

At a minimum the yaml expansion needs to convert human readable start/size fields to numbers.

What they get converted to (i.e. reg property or something else) is dependent on where they are being used.

So in this case, yaml expansion can do some of it and we should add routines to lopper_lib that do specific outputs, i.e. lopper_lib.expand_start_size_to_reg()